### ISSUE TITLE
Tracks without icon show up without name on contribute page, so you don't know which track you're contributing to

### ISSUE THREAD

**Issue by SleeplessByte** [exercism/exercism#6240]
![image](https://user-images.githubusercontent.com/1964376/153755037-61ba75d2-b794-48a8-9076-a06a91de567e.png)

I couldn't come up with a better title, but see the screenshot above.

**Comment by iHiD** [comment]
@ErikSchierboom One for you.

**Comment by ErikSchierboom** [comment]
_TL;DR The icon for the above-mentioned contributions does not show up because the track that was contributed to (`zig`) is not in our database._

I've looked into this, and the reason for this is not that the track (which is `zig`) does not have an icon (because [it does](https://github.com/exercism/website-icons/blob/main/tracks/zig.svg)), but because the track is not in the database. At first I thought we didn't have any inactive tracks in the database, but I was wrong. I've then did a bit of research and scanned all the track repos and checked whether the `active` key in their `config.json` was set to `true` and cross-referenced that against the tracks in our database.

For the active tracks, they all matchup, except for one exception: `z3`. 
This track does have `active` set to `true`, but I don't think it is entirely ready for that. 
For example, its key features are all copied from python and don't really reflect the language. 
There also doesn't seem to be any active maintainer working on it, which is usually a good idea when launching a track.
The reason it is not in the database is that it doesn't sync: https://github.com/exercism/z3/issues/80
That said, the track does seem to have 14 exercises and it is a really interesting track because it is so different from other tracks as it is a theorem solver. 
The slightly weird thing though is that Z3 isn't actually a language but a library, which can be used from many different languages.
So it feels like we should maybe not have created this track at all?

For the inactive tracks, there are more tracks that are not in the database:

- 05ab1e
- ada	
- arm64-assembly	
- babashka
- forth	
- free-pascal	
- gleam
- gnucobol
- io	
- j	
- nix
- qsharp	
- red	
- shen	
- solidity	
- system-verilog	
- vlang	
- wasm	
- zig

Many of them have a track icon, but not all of them. 
For the above bug to be fixed, we could add these tracks to the database and the correct icon would be shown for contributions to these tracks.

While doing this research, I've found that while we don't list inactive tracks on the tracks page, we don't actually guard against inactive tracks being shown: https://exercism.org/tracks/idris 
Is that on purpose @iHiD? I assume it is not, because we probably don't want to show a "Join this track" button for inactive tracks.
If we agree this is not something we want, I could work on a PR to add the relevant guards, or I could do something different. What if we would allow these tracks to be displayed, but with a special banner stating that this track is being developed, possibly with a call to action to contribute to the track?
Having the inactive tracks in the database also allows us to list them in the tasks list.
And we could have a special page listing the tracks that are in the development stage, or maybe add them to the bottom of the tracks page.

**Comment by iHiD** [comment]
> Is that on purpose @iHiD?

Yes. It's for maintainers to test them pre-launch. But we could now guard it so only maintainers can access, as we have that data in the db now. 

I think add them to the db and add the guard so only maintainers can access them. Then once a track is close Tai can make an icon for them. I don't want him having to make lots that don't get used. So before that we need to check there's a placeholder icon that appears in its place.

**Comment by ErikSchierboom** [comment]
@taiyab Could you create a placeholder icon for tracks?

**Comment by taiyab** [comment]
@ErikSchierboom https://github.com/exercism/website-icons/pull/109

I've called the placeholder icon `placeholder.svg`. Don't know if it needs to go into `/placeholders/` instead.

**Comment by ErikSchierboom** [comment]
> I've called the placeholder icon placeholder.svg. Don't know if it needs to go into /placeholders/ instead.

I'm not sure we have a real strategy here. E.g. we have one file in `/placeholders`, but the `website` repo contains `blog-placeholder-article.svg` and `missing-exercise.svg`. @iHiD what would be your preference?

**Comment by iHiD** [comment]
`missing-track.svg`

**Comment by ErikSchierboom** [comment]
@SleeplessByte The missing icon should now appear as @taiyab has added the icons.

To prevent this from happening with future tracks, I've opened https://github.com/exercism/exercism/issues/6240
